test/types/function.test-d.ts

```diff
@@ -0,0 +1,29 @@
+import { expectError, expectNotType, expectType } from 'tsd';
+import App from '../../src/App';
+import type { FunctionCompleteFn, FunctionFailFn } from '../../src/CustomFunction';
+import type { FunctionInputs } from '../../src/types';
+import type { AckFn } from '../../src/types/utilities';
+
+const app = new App({ token: 'TOKEN', signingSecret: 'Signing Secret' });
+
+expectError(
+  app.function('callback', async ({ ack }) => {
+    expectNotType<AckFn<void>>(ack);
```

Comment:
Thanks for adding these tests! Here, I would structure this test (and the other similar sad path tests in this file) like so:

```typescript
// Comment explaining what the intention of the test is, perhaps:
// By default `function` handlers auto-acknowledge events so `ack` should not be provided/defined
expectError(app.function('callback', async ({ ack }) => { }));
```

There's no point testing the type of `ack` within the handler, since the first TS error is trying to even get a reference to `ack` from the dereferenced argument object.

Reply:
Good point I've added the comment 💯 

But I've left `expectNotType<AckFn<void>>(ack);` because typescript complains about `ack is declared but its value is never read.`